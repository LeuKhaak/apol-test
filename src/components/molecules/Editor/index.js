import { React, useState, useEffect } from "react";
import * as styles from "./styles.module.scss";
import Cell from "src/components//molecules/Cell";
import Btn from "src/components/atoms/Btn";

function Editor({ number, data, editeUser, stopEdition, mode }) {
  const saveRowData = (id) => {
    editeUser(id, newData);
    stopEdition();
  };

  const saveRowDataForInput = (event, id) => {
    if (event.key === "Enter") {
      saveRowData(id);
    }
  };

  const shortData = [...data];
  shortData.shift();

  const [newData, setNewData] = useState(data);

  const edit = (ind, event) => {
    const arr = [...newData];
    arr[ind + 1].splice(1, 1, event.target.value);
    setNewData(arr);
  };

  useEffect(() => {
    setNewData(data);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [mode]);

  return (
    <div
      className={styles.editor}
      style={{ display: mode !== "edite" ? "none" : "" }}
    >
      <Cell content={number} cellStyle="number" width="3rem" />
      <div className={styles.formWrapper}>
        <form className={styles.form}>
          {shortData &&
            shortData.map((el, ind) => (
              <div key={ind} className={styles.inputWrapper}>
                <input
                  name={el[0]}
                  value={newData[ind + 1][1]}
                  onChange={(event) => edit(ind, event)}
                  onKeyPress={(event) => saveRowDataForInput(event, data[0][1])}
                />
              </div>
            ))}
        </form>
      </div>
      <div className={styles.actions}>
        <Btn
          id={number}
          btnText="Save"
          btnStyle="edite"
          onClick={() => saveRowData(data[0][1])}
        />
        <Btn id={number} btnText="Delete" btnStyle="inactive" />
      </div>
    </div>
  );
}

export default Editor;
