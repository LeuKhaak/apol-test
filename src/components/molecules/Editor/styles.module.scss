@import "../../../assets/styles/styles.scss";
.editor {
  display: flex;
  display: grid;
  grid-template-columns: 4rem 58rem 15rem;
  @include desktop {
    grid-template-columns: 3rem 58rem 8rem;
  }
  @include desktopSmall {
    grid-template-columns: 3rem 58rem;
  }
  @include tabletLandscape {
    grid-template-columns: 2rem 48rem;
  }
  @include tabletSmall {
    grid-template-columns: 2rem 44.5rem;
  }
  @include tabletSmallSecond {
    grid-template-columns: 2rem 39rem;
  }
  @include mobile {
    grid-template-columns: 2rem 35rem;
  }
  @include mobileSecond {
    grid-template-columns: 2rem 31rem;
  }
  @include mobileSmall {
    grid-template-columns: 2rem 27rem;
  }
  @include mobileSmallSecond {
    grid-template-columns: 2rem 24.5rem;
  }
  @include mobileSmallThird {
    grid-template-columns: 1.85rem 18rem;
  }
  @include mobileSmallest {
    display: flex;
    flex-wrap: wrap;
  }
  .formWrapper {
    .form {
      display: grid;
      grid-template-columns: 10rem 7rem 7rem 10rem 10rem 7rem 7rem;
      @include tabletLandscape {
        grid-template-columns: 8rem 6rem 6rem 8rem 8rem 6rem 6rem;
      }
      @include tabletSmall {
        grid-template-columns: 7.5rem 5.5rem 5.5rem 7.5rem 7.5rem 5.5rem 5.5rem;
      }
      @include tabletSmallSecond {
        grid-template-columns: 7rem 5rem 3rem 7rem 7rem 5rem 5rem;
      }
      @include mobile {
        grid-template-columns: 7rem 4.5rem 3rem 6.5rem 6.5rem 4rem 3.5rem;
      }
      @include mobileSecond {
        grid-template-columns: 6rem 3.5rem 3rem 5.5rem 5.5rem 4rem 3.5rem;
      }
      @include mobileSmall {
        grid-template-columns: 5rem 3.5rem 3rem 4rem 4rem 4rem 3.5rem;
      }
      @include mobileSmallSecond {
        grid-template-columns: 4rem 3rem 3rem 3.5rem 3.5rem 4rem 3.5rem;
      }
      @include mobileSmallThird {
        grid-template-columns: 3rem 2.5rem 2.5rem 3rem 3rem 3.75rem 3.25rem;
      }
      @include mobileSmallest {
        display: flex;
        flex-wrap: wrap;
      }
      //.inputWrapper {
      //  border: solid 0.1rem $greyBorderColor;
      //  padding: 0.5rem;
      //  input {
      //    max-width: calc(100% - 0.5rem);
      //    padding: 0.5rem 0.25rem;
      //    min-height: 1rem;
      //    font-weight: 400;
      //  }
      //}
    }
  }
  .actions {
    @include cell;
    display: flex;
  }
}
