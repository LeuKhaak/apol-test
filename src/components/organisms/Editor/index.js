import React, { useState, useEffect } from "react";
import * as styles from "./styles.module.scss";
import Cell from "../../molecules/Cell";
import Input from "../../molecules/Input";
import Btn from "../../atoms/Btn";

function Editor({ number, data, editUser, stopEdition, mode }) {
  const saveRowData = (id) => {
    editUser(id, newData);
    stopEdition();
  };

  const saveRowDataForInput = (event, id) => {
    if (event.key === "Enter") {
      saveRowData(id);
    }
  };

  const shortData = [...data];
  shortData.shift();

  const [newData, setNewData] = useState(data);

  const edit = (ind, event) => {
    const arr = [...newData];
    arr[ind + 1].splice(1, 1, event.target.value);
    setNewData(arr);
  };

  useEffect(() => {
    setNewData(data);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [mode]);

  return (
    <div
      className={styles.editor}
      style={{ display: mode !== "edit" ? "none" : "" }}
    >
      <Cell content={number} cellStyle="number" width="3rem" />
      <div className={styles.formWrapper}>
        <form className={styles.form}>
          {shortData &&
            shortData.map((el, ind) => (
              <Input
                key={ind}
                ind={ind}
                el={el}
                userId={data[0][1]}
                value={newData[ind + 1][1]}
                edit={edit}
                saveRowDataForInput={saveRowDataForInput}
              />
            ))}
        </form>
      </div>
      <div className={styles.actions}>
        <Btn
          id={data[0][1]}
          btnText="Save"
          btnStyle="edit"
          onClick={saveRowData}
        />
        <Btn id={number} btnText="Delete" btnStyle="inactive" />
      </div>
    </div>
  );
}

export default Editor;
