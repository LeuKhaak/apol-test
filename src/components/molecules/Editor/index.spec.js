import React from "react";
import { shallow } from "enzyme";
import Editor from ".";

let component = null;
const onChange = jest.fn();
const data = [
  ["a", "b"],
  ["c", "d"],
  ["e", "f"],
  ["g", "h"],
];
const setComponent = (props) => shallow(<Editor {...props} />);
beforeEach(() => {
  component = setComponent();
});

describe("Editor testing:", () => {
  test("is Editor rendered", () => {
    component = setComponent({
      mode: "edite",
      data: data,
    });
    expect(component).toMatchSnapshot();
  });

  //test("check onChangeFunc", () => {
  //  component = setComponent({
  //    onChange: onChange,
  //    mode: "edite",
  //    data: data,
  //  });
  //  component.find("input").simulate(("change", { target: { value: "S" } }));
  //  expect(onChange).toHaveBeenCalled();
  //});
});
