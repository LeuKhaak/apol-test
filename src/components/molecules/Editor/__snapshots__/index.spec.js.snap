// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor testing: is Editor rendered 1`] = `
<div
  style={
    Object {
      "display": "none",
    }
  }
>
  <Cell
    cellStyle="number"
    width="3rem"
  />
  <div>
    <form>
      <Input
        edit={[Function]}
        el={
          Array [
            "c",
            "d",
          ]
        }
        ind={0}
        key="0"
        saveRowDataForInput={[Function]}
        userId="b"
        value="d"
      />
      <Input
        edit={[Function]}
        el={
          Array [
            "e",
            "f",
          ]
        }
        ind={1}
        key="1"
        saveRowDataForInput={[Function]}
        userId="b"
        value="f"
      />
      <Input
        edit={[Function]}
        el={
          Array [
            "g",
            "h",
          ]
        }
        ind={2}
        key="2"
        saveRowDataForInput={[Function]}
        userId="b"
        value="h"
      />
    </form>
  </div>
  <div>
    <Btn
      btnStyle="edit"
      btnText="Save"
      id="b"
      onClick={[Function]}
    />
    <Btn
      btnStyle="inactive"
      btnText="Delete"
    />
  </div>
</div>
`;
